src/util/parse-package-path.js

```diff
@@ -0,0 +1,26 @@
+/* @flow */
+
+const PACKAGE_SEP = '/';
+
+/**
+ * Parse input strings like `package-1/package-2` to an array of packages
+ */
+export default function parsePackagePath(input: string): string[] {
+  let prevScoped = false
+  let packages = []
+  input.split(PACKAGE_SEP).forEach(pkg => {
+    if (prevScoped) {
+      // If package is scoped, then we should concat it's name to the prev part
+      packages[packages.length - 1] += PACKAGE_SEP + pkg;
+      prevScoped = false
+    } else {
+      packages.push(pkg);
+    }
+
+    if (pkg.indexOf('@') !== -1) {
+      prevScoped = true;
+    }
+  });
+
+  return packages
```

Comment:
Ok. I thought about it but came up with functional version. These parsing regexps rapidly becomes unreadable 😄.